rewrite-gradle/src/main/java/org/openrewrite/gradle/AddDependencyVisitor.java

```diff
@@ -85,51 +88,90 @@ public class AddDependencyVisitor extends GroovyIsoVisitor<ExecutionContext> {
-                    g.getStatements().isEmpty() ?
-                            dependenciesInvocation :
-                            dependenciesInvocation.withPrefix(Space.format("\n\n"))));
-        }
+            if (dependenciesBlockMissing) {
+                if (cu instanceof G.CompilationUnit) {
+                    G.CompilationUnit g = (G.CompilationUnit) cu;
+                    Statement dependenciesInvocation = GRADLE_PARSER.parse(ctx, "dependencies {}")
+                            .findFirst()
+                            .map(G.CompilationUnit.class::cast)
+                            .orElseThrow(() -> new IllegalArgumentException("Could not parse as Gradle"))
+                            .getStatements().get(0);
+                    Cursor parent = getCursor();
+                    setCursor(new Cursor(parent, g));
+                    dependenciesInvocation = autoFormat(dependenciesInvocation, ctx, new Cursor(parent, g));
+                    setCursor(parent);
+                    cu = g.withStatements(ListUtils.concat(g.getStatements(),
+                            g.getStatements().isEmpty() ?
+                                    dependenciesInvocation :
+                                    dependenciesInvocation.withPrefix(Space.format("\n\n"))));
+                } else {
+                    K.CompilationUnit k = (K.CompilationUnit) cu;
+                    J.MethodInvocation dependenciesInvocation = (J.MethodInvocation) ((J.Block) GRADLE_PARSER.parseInputs(Collections.singletonList(new GradleParser.Input(Paths.get("build.gradle.kts"), () -> new ByteArrayInputStream("dependencies {}".getBytes(StandardCharsets.UTF_8)))), null, ctx)
+                            .findFirst()
+                            .map(K.CompilationUnit.class::cast)
+                            .orElseThrow(() -> new IllegalArgumentException("Could not parse as Gradle"))
+                            .getStatements().get(0)).getStatements().get(0);
+                    Cursor parent = getCursor();
+                    setCursor(new Cursor(parent, k));
+                    dependenciesInvocation = autoFormat(dependenciesInvocation, ctx, new Cursor(getCursor(), k));
+                    setCursor(parent);
+                    dependenciesInvocation = dependenciesInvocation.withArguments(ListUtils.mapFirst(dependenciesInvocation.getArguments(), arg -> {
+                        J.Lambda lambda = (J.Lambda) requireNonNull(arg);
+                        J.Block block = (J.Block) lambda.getBody();
+                        return lambda.withBody(block.withEnd(Space.format("\n")));
+                    }));
+                    cu = k.withStatements(ListUtils.concat(k.getStatements(),
+                            k.getStatements().isEmpty() ?
+                                    dependenciesInvocation :
+                                    dependenciesInvocation.withPrefix(Space.format("\n\n"))));
```

Comment:
Is this something that would get simpler with `KotlinTemplate`?

Reply:
Yep, it is simpler. I've got a small POC for it locally, but hadn't yet committed to making the swap. If you think it'd be better to see that task completed now, I'm willing to commit to making that additional update.

Reply:
From my POV this PR is good to merge, as it is.